packages/flutter/lib/src/widgets/autocomplete.dart

```diff
@@ -707,7 +711,7 @@ class _RawAutocompleteState<T extends Object> extends State<RawAutocomplete<T>>
         },
       );
       Overlay.of(context, rootOverlay: true)!.insert(_floatingOptions!);
-    } else if (_floatingOptions != null) {
+    } else if ((_floatingOptions != null && !_focusNode.hasFocus) || (_selection != null && _focusNode.hasFocus)) {
```

Comment:
I'm not sure how to have the overlay continue to show if the overlay itself receives focus for keyboard traversal. I'm using https://material-ui.com/components/autocomplete/ as a reference for how this should behave.